example/example.script

```diff
@@ -39,12 +39,13 @@ function on_input(self, action_id, action)
 			
 			self.click_num = self.click_num + 1
 		else
-			local png, w, h = screenshot.png()
+			local x, y, width, height = 100, 300, 500, 600
+			local png, w, h = screenshot.png(x, y, width, height)
 			local f = io.open(sys.get_save_file(sys.get_config("project.title", "screenshot"), "screenshot1.png"), "wb")
 			f:write(png)
 			f:flush()
 			f:close()
-			local buffer, w, h = screenshot.buffer()
+			local buffer, w, h = screenshot.buffer(x, y, width, height)
 			resource.set_texture(go.get("cube#model", "texture0"), { type = resource.TEXTURE_TYPE_2D, width = w, height = h, format = resource.TEXTURE_FORMAT_RGBA }, buffer)
```

Comment:
With Vulkan it doesn't work correctly. Appears following log
```
dmengine[79953:38968550] [CAMetalLayerDrawable texture] should not be called after already presenting this drawable. Get a nextDrawable instead.
```

Reply:
Problem is not in `ReadPixels` calls. If I make two sequential calls and save result to the file - screenshots look correctly. But in case of read to buffer and set it as texture - empty texture displayed.

Reply:
That's odd! Not sure why, but let's maybe leave that for later?

Reply:
Fixes on engine's side https://github.com/defold/defold/pull/10420